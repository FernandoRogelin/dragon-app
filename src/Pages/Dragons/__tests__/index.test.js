import React from "react";
import { render, act, waitForElement } from "@testing-library/react";
import { Server } from "miragejs";

import Dragon from "..";

const promise = Promise.resolve();

let server;

describe("Dragon list component", () => {
  beforeEach(() => {
    jest.clearAllMocks();
    server = new Server();
  });

  afterEach(() => {
    server.shutdown();
  });

  test("Snapshot test", async () => {
    const { container, getByTestId } = render(<Dragon />);

    server.get(
      "http://5c4b2a47aa8ee500142b4887.mockapi.io/api/v1/dragon",
      () => [{ id: 1, name: "Drake", type: "Orange", createdAt: "20/09/2020" }]
    );

    await waitForElement(() => getByTestId("items"));

    expect(getByTestId("nameDragon")).toHaveTextContent("Drake");

    expect(container.firstChild).toMatchSnapshot();

    await act(() => promise);
  });
});
