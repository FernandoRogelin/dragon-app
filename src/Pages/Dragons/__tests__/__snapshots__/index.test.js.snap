// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dragon list component Snapshot test 1`] = `
<div
  class="styles__WrapperComponent-mms3ln-0 gZmSTy"
>
  <h1
    class="styles__Title-mms3ln-1 wrRRR"
  >
    Timeline de Dragões
  </h1>
  <button
    class="Button__button-a83c5q-0 kzEeEq"
  >
    Adicionar
  </button>
  <div
    class="styles__Content-sc-6cu4uz-0 knxsRm styles__Table-sc-11yc90o-0 kwrvqQ"
  >
    <table
      class="styles__TableComponent-sc-6cu4uz-1 djUGph"
    >
      <thead
        class="styles__Thead-sc-6cu4uz-2 RsIOg"
      >
        <tr
          class="styles__Tr-sc-6cu4uz-3 kXRpur"
        >
          <th
            class="styles__Th-sc-6cu4uz-4 fsOwmb"
          >
            Dragão
          </th>
          <th
            class="styles__Th-sc-6cu4uz-4 fsOwmb"
          >
            Ações
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="styles__Tr-sc-11yc90o-1 boAmly"
          data-testid="items"
        >
          <td
            class="styles__Td-sc-11yc90o-2 iBZGcK"
            data-testid="nameDragon"
          >
            Drake
          </td>
          <td
            class="styles__Td-sc-11yc90o-2 iBZGcK"
          >
            <button
              class="Button__button-a83c5q-0 kzEeEq"
            >
              Editar
            </button>
            <button
              class="Button__button-a83c5q-0 kzEeEq"
            >
              Detalhes
            </button>
            <button
              class="Button__button-a83c5q-0 kzEeEq"
              data-testid="deleteDragon"
            >
              Deletar
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
